eLearning-courseware/eLearning.jsonld

```diff
@@ -0,0 +1,1271 @@
+        "type": "StatementTemplate",
+        "inScheme": "https://w3id.org/xapi/eLearning",
+        "prefLabel": {
+          "en": "Page launched"
+        },
+        "definition": {
+          "en": "Learner has launched the page."
+        },
+        "rules": [
+          {
+            "location": "$.actor.mbox",
+            "presence": "included"
+          },
+          {
+            "location": "$.verb.id",
+            "any": ["http://adlnet.gov/expapi/verbs/launched"]
+          },
+          {
+            "location": "$.object.id",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.name",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.description",
+            "presence": "included"
+          },
+          {
+            "location": "$.context.extensions.pageId",
+            "presence": "recommended"
+          },
+          {
+            "location": "$.context.extensions.topicId",
+            "presence": "recommended"
+          }
+        ]
+      },
+      {
+        "id": "https://github.com/TryxAPI/xAPI-eLearning-Courseware-Profile/templates/pageLeft",
+        "type": "StatementTemplate",
+        "inScheme": "https://w3id.org/xapi/eLearning",
+        "prefLabel": {
+          "en": "Page left"
+        },
+        "definition": {
+          "en": "Learner has left the page."
+        },
+        "rules": [
+          {
+            "location": "$.actor.mbox",
+            "presence": "included"
+          },
+          {
+            "location": "$.verb.id",
+            "any": ["http://adlnet.gov/expapi/verbs/progressed"]
+          },
+          {
+            "location": "$.object.id",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.name",
+            "presence": "included"
+          },
+          {
+            "location": "$.object.definition.description",
+            "presence": "included"
+          },
+          {
+            "location": "$.context.extensions.pageId",
+            "presence": "recommended"
+          },
+          {
+            "location": "$.context.extensions.topicId",
+            "presence": "recommended"
+          }
+        ]
+      },
+      {
+        "id": "https://github.com/TryxAPI/xAPI-eLearning-Courseware-Profile/templates/topicCompleted",
+        "type": "StatementTemplate",
+        "inScheme": "https://w3id.org/xapi/eLearning",
+        "prefLabel": {
+          "en": "Topic completed"
+        },
+        "definition": {
+          "en": "Learner has completed the topic."
+        },
+        "rules": [
+          {
+            "location": "$.actor.mbox",
+            "presence": "included"
+          },
+          {
+            "location": "$.verb.id",
+            "any": ["http://adlnet.gov/expapi/verbs/completed"]
```

Comment:
I notice you're repeating the use of completed as the only determining property here; it looks like the rules are the same, so in that case you could have a single statement template covering both completion cases, or alternatively you could add another determining property to each to differentiate them, probably objectActivityType

Reply:
Good idea. This was done for `launched` and `progressed` templates for now.